bin/run_multiple_panels.py

```diff
@@ -0,0 +1,507 @@
+        if tech2 == 'Clothes Dryer':
+            n_delay, N, t_delay, deltat_delay, n_pop = dryer_control(sim_type, dwelling, P_prim, t, pipeline, n_delay, N, t_delay, deltat_delay, n_pop)
+        else:        
+            if P_prim > 0:
+                if tech2 == 'Water Heating':
+                    control_signal = {tech2: {'Load Fraction': 0}}
+                else: # EV
+                    control_signal = {tech2: {'P Setpoint': 0}}
+                N = N+1
+                t_delay.append(dwelling.current_time)
+            else:
+                # keep previous control signal (same as forward fill)
+                control_signal = None
+
+    df = pd.DataFrame(t_delay, columns=['Timestamp'])
+    if tech2 == 'Clothes Dryer':
+        df['Delayed Time (s)'] = deltat_delay
+    df.to_csv(os.path.join(output_path, tech2+'_metrics.csv'), index=False)
+
+    dwelling.finalize()
+    
+    
+def record_and_remove_cycle(dwelling, schedule, t, pipeline):
+    
+    # all zero indicies
+    zero_indices = schedule.index[schedule == 0].tolist()
+    
+    # find the last index that the power is zero - cycle start
+    cycle_start = max([index for index in zero_indices if index < t], default=None) + dwelling.time_res
+        
+    # find the next index that the power is zero - cycle end
+    cycle_end = min([index for index in zero_indices if index > t], default=None) - dwelling.time_res
+    
+    # record cycle in pipeline
+    pipeline.append(schedule[cycle_start:cycle_end].copy())
+    
+    # remove cycle from schedule
+    for i in pd.date_range(cycle_start, cycle_end, freq=dwelling.time_res, inclusive='both'):
+        schedule.loc[i] = 0
+    
+    return(schedule)
+
+
+def add_first_cycle_to_schedule(dwelling, schedule, pipeline, deltat_delay, n_pop):
+    
+    # first cycle in pipeline
+    cycle_start = pipeline[0].index[0]
+    cycle_end = pipeline[0].index[-1]
+    cycle_length = cycle_end - cycle_start
+    
+    # add to schedule
+    schedule[dwelling.current_time:dwelling.current_time + cycle_length] = pipeline[0]
+        
+    # update pipeline
+    deltat_delay[n_pop] = dwelling.current_time - deltat_delay[n_pop]
+    pipeline.pop(0)
+    n_pop += 1
+    
+    return(schedule, n_pop)
+
+
+def coincide_with_next_cycle(dwelling, schedule, pipeline):
+    
+    cycle_start = pipeline[0].index[0]
+    cycle_end = pipeline[0].index[-1]
+    cycle_length = cycle_end - cycle_start
+    
+    for i in pd.date_range(dwelling.current_time, dwelling.current_time + cycle_length, freq=dwelling.time_res, inclusive='both'):
+        if schedule.loc[i] > 0:
+            return(True, i)
+            
+    return(False, dwelling.current_time)
+
+
+def exceed_threshold(P_prim, size, pipeline):
+    
+    if P_prim + pipeline[0][0] > 0.8*size*240/1000:
+        return(True)
+    else:
+        return(False)
+
+
+def dryer_control(sim_type, dwelling, P_prim, t, pipeline, n_delay, N, t_delay, deltat_delay, n_pop, size=0):
```

Comment:
I'm a little torn on what to keep. I think there's a lot of good code that's project-specific that probably shouldn't get merged in. I think we only want to keep code that provides useful examples of what others may want to do. I'd suggest removing all the clothes dryer code since it's too specific to this use case.

Reply:
It's a bit specific, sure, but couldn't we also generically apply a lot of this to any appliance where we're worried about pausing it and having to restart a cycle?

Like for panels, I was thinking a generic version of this might be one of the controls we have to consider, where the smart panels turns off x device to avoid overload the panel, and then that cycle has to restart when other things have turned off and there's sufficient capacity to try to restart x and have it complete a full cycle.

Reply:
Let me know if I should keep the dryer control function. I think it's a good example for controlling cycle-based appliances such as washers and dryers.

Reply:
Resolution: Longer term, we'll have event based appliances, and so we probably don't need to keep this specific example. We'll always have it backed up on the original PR, and as part of the new panels project we'll be changing appliance modeling to make this way easier and avoid this kind of workaround approach.

Reply:
Sounds good. I removed dryer control from the script.